src/Pages/RoomList/PageButtons/PageButtons.js

```diff
@@ -0,0 +1,124 @@
+import React from "react";
+import styled from "styled-components";
+import { flexCenter } from "../../../styles/Theme";
+
+const PageButtons = ({ fetchData, homesCount }) => {
+  return (
+    <Pagination>
+      <PageBtnContainer>
+        <PrevPage className="noClick" />
+
+        <PageBtn data-index="1" onClick={fetchData}>
+          1
+        </PageBtn>
+        <PageBtn data-index="2" onClick={fetchData}>
+          2
+        </PageBtn>
+        <PageBtn data-index="3" onClick={fetchData}>
+          3
+        </PageBtn>
+        <PageBtn data-index="4" onClick={fetchData}>
+          4
+        </PageBtn>
+        <PageBtn data-index="5" onClick={fetchData}>
+          5
+        </PageBtn>
+        <NextPage />
+      </PageBtnContainer>
```

Comment:
```suggestion
      <PageBtnContainer onClick={fetchData}>
        <PrevPage className="noClick" />
        {[...Array(5)].map((_, idx) => (
          <PageBtn data-index={idx + 1}>{idx + 1}</PageBtn>
        ))}
        <NextPage />
      </PageBtnContainer>
```
- 반복되는 데이터는 배열 타입으로 만들어서 시도해보시기 바랍니다.
- `[...Array(5)]` 와 같이 Array 생성자를 사용하여 요소가 5개인 배열을 생성할 수 있습니다.
- 콜백 함수의 첫 번째 인자를 사용 안 하는 경우 `_` 이렇게 많이 사용합니다.
- <PageBtn> 각각에 이벤트 리스너를 달지 않고 상위 요소에서 처리해줄 수 있습니다. (cf. 이벤트 전파)
```js
  const fetchData = e => {
    const offset = e?.currentTarget.dataset.index;
    if (!offset) return;
    fetch(`${API}/homes?limit=${LIMIT}&offset=${offset * LIMIT}`)
      .then(res => res.json())
      .then(data => setHomes(data.homes));
  };
```
- props로 전달받는 fetchData 함수입니다. (참고로 함수명 수정해주세요!!!)
- `e.target` vs. `e.currentTarget` 에 대해 학습해보시기 바랍니다. 간략한 설명 댓글로 달아주세요!
- 부모 요소에 이벤트를 전파 했기 때문에, <PrevPage> 혹은 <NextPage> 에서 이벤트가 일어난 경우 offset이 존재 하지 않을 수 있기 때문에 예외처리 해주었습니다.
- 질문) `e`에 optional chaining 적용해주신 이유가 있나요?

Reply:
와진짜 넘감사합니다 Array생성자 문법이 있군요... 너무 멋있고 깔끔해졌어요 감사합니다!!!!!

Reply:
e에 optional chaining 적용한 건.. 페이지네이션 세션을 그대로 따라하다가 남은 것입니다..! ㅋㅋㅋ보니까 역시 필요없네요 지우겠습니다ㅠㅠ

Reply:
은진님! 제가 반대로 생각하고 있었네요 target이 맞습니다🙌 (ㅈㅅㅈㅅ)